### ISSUE TITLE
Special characters in variable names cause errors

### ISSUE THREAD

**Issue by teetotum** [MakhBeth/js-to-scss#1]
We have variables with special characters in their names...
```
primary: {
    main: '#E6B233',
    dark: '#58A818',
    light: '#6ECE3E',
    '60%': '#233E6B99',
    '30%': '#233E6B4C',
    contrastText: '#FFFFFF',
  },
```
...which will cause errors when processed by `js-to-scss`. So currently we just copied the transformation function from your library locally, modified it to escape special chars, and use the modified function in our webpack config.

Would you accept a PR for this?

It currently only cares for and escapes `%` and would probably need some more love before being ready for release in a lib.

I just add the modified code here, as a base for discussion.

```ts
const escapeSpecialChars = (string: any) => string.replaceAll('%', '\\%');

export const jsToScss = (obj: any, prefix = '$', transform = (_derivedKey: string, val: any) => val): string => {
  return Object.entries(obj).reduce((accumulated, [key, value]) => {
    const derivedKey = `${prefix}${escapeSpecialChars(key)}`;
    if (typeof value === 'object' && !Array.isArray(value) && value) {
      const processedObject = jsToScss(value, `${derivedKey}-`, transform);
      return accumulated + `${processedObject}`;
    } else {
      const processedValue = Array.isArray(value) ? `(${transform(derivedKey, value)})` : transform(derivedKey, value);
      return accumulated + `${derivedKey}: ${processedValue}; `;
    }
  }, '');
};
```

**Comment by MakhBeth** [comment]
Sorry for the late answer. Yes, I think that the only use case could be a percentage. I don't think that there is any other use case. 

My question is: How are you using those variables in your Sass? With the escape? Can you provide an example?

Maybe a more “Sassy” way should be meta.inspect. But it's only meant to debug. So
I would rely on JS

PR are welcomed btw

**Comment by teetotum** [comment]
The usage in SCSS looks like this:
```scss
.foobar {
  background-color: $primary-60\%;
}
```
My notion is that `js-to-scss` could escape any characters that would otherwise lead to syntactically invalid sass variable names (more about that: [stackoverflow](https://stackoverflow.com/questions/17191265/legal-characters-for-sass-and-scss-variable-names) and [blog article](https://vanseodesign.com/css/how-to-use-sass-variables/#:~:text=Allowed%20Characters%20in%20Variable%20Names&text=Any%20number%200%E2%80%939%20(as,if%20escaped%20with%20a%20backslash. )).
As I'm not aware of any utility function of the sass lib to help with that, I reckon we could just work with a hard-coded list of rules regarding allowed characters (that mirrors the rules of the sass compiler) and escape anything that isn't covered by the rules. Would be good enough I believe.

**Comment by MakhBeth** [comment]
Merged in #2
